design/design-doc-observe-only-resources.md

```diff
@@ -0,0 +1,703 @@
+won’t be possible to create an observe-only resource without providing a value
+for any required fields, because the Kubernetes API checks for the presence of
+these fields before allowing the CR to be created.
+
+> Please note this is already the case with [importing existing managed resources]
+> today.
+
+We will fix this by leveraging the [Common Expression Language (CEL)], which was
+graduated to beta (i.e. enabled by default) as of Kubernetes 1.25. See the
+following diff for the required changes we need for making [CIDRBlock] parameter
+of AWS VPC required only if not `ObserveOnly`:
+
+```diff
+		// CIDRBlock is the IPv4 network range for the VPC, in CIDR notation. For
+        // example, 10.0.0.0/16.
+-       // +kubebuilder:validation:Required
+        // +immutable
+-       CIDRBlock string `json:"cidrBlock"`
++       CIDRBlock *string `json:"cidrBlock,omitempty"`
+
+        // The IPv6 CIDR block from the IPv6 address pool. You must also specify Ipv6Pool
+        // in the request. To let Amazon choose the IPv6 CIDR block for you, omit this
+@@ -170,6 +169,7 @@ type VPC struct {
+        metav1.TypeMeta   `json:",inline"`
+        metav1.ObjectMeta `json:"metadata,omitempty"`
+
++       // +kubebuilder:validation:XValidation:rule="self.managementPolicy == 'ObserveOnly' || has(self.forProvider.cidrBlock)",mess
+age="cidrBlock is a required parameter"
+        Spec   VPCSpec   `json:"spec"`
+        Status VPCStatus `json:"status,omitempty"`
+ }
+```
+
+#### New Import Procedure
+
+The [import procedure][import existing managed resources] documented today has
+some limitations and caveats as follows:
+
+1. Users need to provide all the required fields in the spec of the resource
+   with correct values even though they are not used for importing the resource.
+   A wrong value for a required field will result a configuration update which
+   is not desired.
+2. Any typo in the external name annotation or some mistake in the identifying
+   arguments (e.g. `region`) will result creation of a new resource instead of
+   importing the existing one.
+
+While it is not directly related to this proposal, we will also address these
+issues by introducing a new import procedure that will be made available with
+the new `ObserveOnly` policy. The new procedure will be as follows:
+
+1. Create a new resource with `ObserveOnly` policy.
+   1. With external name annotation set to the external name of the resource to be imported.
+   2. Only provide the identifying arguments (e.g. `region`) in the spec of the
+      resource and skip all the other fields including the required ones ( which 
+      would no longer be required, see the previous section).
+2. Expect the existing resource to be observed successfully indicating that the
+   existing resource is found.
+3. Change the policy to `Full` and provide the required fields by copying them
+   from `status.atProvider` to give full control of the resource to Crossplane.
```

Comment:
I wonder if we could automate this.

If `spec.managementPolicy` is `Full`, `spec.forProvider` is empty, and `status.atProvider` is not Crossplane automatically copies everything (that can be changed) from `status.atProvider` to `spec.forProvider`.

Perhaps this would be too implicit/surprising?

Reply:
Yes, I believe this could be automated as you suggested. We wouldn't even need everything but only the required fields and late-init would take care of the rest as usual. It would be great if we could even implement it at managed reconciler level (i.e. in a common place). 

> Perhaps this would be too implicit/surprising?

I don't think so. But we might be more confident after hearing some real use cases or need for this.

Reply:
FWIW, this is the same conclusion we arrived at this morning when we took the OO feature for an early preview test drive :)

We first created a OO resource by specifying the external name annotation, the region, and managementPolicy of ObserveOnly.  Everything got synced successfully and status.atProvider was populated fully. We were hoping to complete the import by then just flipping managementPolicy to FullControl and have the controller take care of everything for us to update the spec to match the reality we had already observed.

So +1 data point for expectations being that the controller would automate this for us 🤪

Reply:
Opened an issue to track this: https://github.com/crossplane/crossplane/issues/3999